### PR TITLE
Support content blocking

### PR DESCRIPTION
**Pull request by nox**
This is @jdm's rebased work from https://github.com/servo/servo/issues/9749#issuecomment-211115746 with a small addition with regard to cross-origin blocking. Mainly opening this for review and for further discussions.

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/11413)

### CODE REVIEW

**Comment by jdm** [comment]
We should leave a TODO in the final commit, since we should use the actual request origin when it's available rather than inferring it via the referrer.

**Comment by nox** [comment]
@paulrouget wants a flag for it (#11500) but as long as we don't ship any builtin blocking rules it shouldn't matter to get this in as such.

File: components/net/content_blocker.rs [hunk 1]
```diff
@@ -0,0 +1,33 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#![allow(unsafe_code)]
```
Comment (Ms2ger) [inline components/net/content_blocker.rs]: You should be able to remove this.

**Comment by nox** [comment]
@bors-servo r=nox+jdm+Ms2ger

Forgot to account for the Url change in content-blocker 0.2.0...

**Comment by nox** [comment]
@bors-servo r=nox+jdm+Ms2ger

Fixed the unit tests.

**Pull request merged**